### PR TITLE
threads: some cleanup

### PR DESCRIPTION
**Pull request by Megamouse**
- Replace magic number id_type with thread_class
- Initialize some uninitialized rsx thread members
- Fix spu_thread audit check
- Add comments to reference old scheduler at least somewhere in the code

### CODE REVIEW

File: rpcs3/Emu/CPU/CPUThread.h [hunk 1]
```diff
@@ -296,6 +301,7 @@ class cpu_thread
 
 private:
 	static thread_local cpu_thread* g_tls_this_thread;
+	const thread_class m_class;
```
Comment (elad335) [inline rpcs3/Emu/CPU/CPUThread.h:304]: Dont add or remove any member of cpu_thread, existing PPU LLVM cache breaks with this change.

**Review by elad335 — CHANGES_REQUESTED** [review]

File: rpcs3/Emu/CPU/CPUThread.h (hunk 1 above)
Comment (Megamouse) [reply rpcs3/Emu/CPU/CPUThread.h:304]: then the cache seems to need a redesign

Comment (elad335) [reply rpcs3/Emu/CPU/CPUThread.h]: Maybe, but that has nothing to do with this pr though. You can simply assign enum values to the previously magic numbers and call it a day.

File: Utilities/Thread.h [hunk 2]
```diff
@@ -36,6 +36,17 @@ enum class thread_state : u32
 	mask = 3
 };
 
+static inline thread_class get_thread_class(u32 thread_id_type)
+{
+	switch (thread_id_type)
+	{
+	case 1: return thread_class::ppu;
```
Comment (elad335) [inline Utilities/Thread.h:43]: This function can replaced with a simple cast.
thread_class::ppu = 1, thread_class::spu = 2, thread_class::rsx = 0x55

I dont remember where but its used in performance-sensituve context.

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 3]
```diff
@@ -5319,7 +5319,7 @@ s64 spu_thread::get_ch_value(u32 ch)
 						atomic_wait_engine::set_one_time_use_wait_callback(+[](u64) -> bool
 						{
 							const auto _this = static_cast<spu_thread*>(cpu_thread::get_current());
-							AUDIT(_this->id_type() == 1);
+							AUDIT(_this->get_class() == thread_class::ppu);
```
Comment (elad335) [inline rpcs3/Emu/Cell/SPUThread.cpp:5322]: Which was a bug, spu type should be used here.

File: rpcs3/Emu/Io/GunCon3.cpp [hunk 4]
```diff
@@ -169,7 +169,7 @@ void usb_device_guncon3::interrupt_transfer(u32 buf_size, u8* buf, u32 endpoint,
 
 	if ((endpoint & 0x80) == LIBUSB_ENDPOINT_OUT)
 	{
-		std::memcpy(m_key, buf, std::min(buf_size, static_cast<u32>(sizeof(m_key))));
+		std::memcpy(m_key.data(), buf, std::min<u32>(buf_size, m_key.size()));
```
Comment (elad335) [inline rpcs3/Emu/Io/GunCon3.cpp:172]: `min<usz>`

File: rpcs3/util/serialization_ext.cpp [hunk 5]
```diff
@@ -1253,7 +1251,6 @@ void compressed_zstd_serialization_file_handler::stream_data_prepare_thread_op()
 			break;
 		}
 
-		usz buffer_offset = 0;
```
Comment (elad335) [inline rpcs3/util/serialization_ext.cpp:1256]: Remove whitespace too

File: rpcs3/rpcs3qt/memory_viewer_panel.cpp [hunk 6]
```diff
@@ -41,9 +41,9 @@ memory_viewer_panel::memory_viewer_panel(QWidget* parent, std::shared_ptr<CPUDis
 		const auto cpu = m_get_cpu();
 
 		if (!cpu) return thread_type::none;
-		if (cpu->id_type() == 1) return thread_type::ppu;
-		if (cpu->id_type() == 0x55) return thread_type::rsx;
-		if (cpu->id_type() == 2) return thread_type::spu;
+		if (cpu->get_class() == thread_class::ppu) return thread_type::ppu;
+		if (cpu->get_class() == thread_class::rsx) return thread_type::rsx;
+		if (cpu->get_class() == thread_class::spu) return thread_type::spu;
```
Comment (elad335) [inline rpcs3/rpcs3qt/memory_viewer_panel.cpp:46]: Call it once and save the value.

**Review by elad335 — CHANGES_REQUESTED** [review]

File: rpcs3/util/serialization_ext.cpp (hunk 5 above)
Comment (Megamouse) [reply rpcs3/util/serialization_ext.cpp:1256]: lol what?

Comment (elad335) [reply rpcs3/util/serialization_ext.cpp:1256]: You didnt remove the newline, there are 2 consequives newlines.

Comment (Megamouse) [reply rpcs3/util/serialization_ext.cpp:1256]: What are you talking about?

Comment (elad335) [reply rpcs3/util/serialization_ext.cpp:1256]: lol, just look at the code here at your text editor.

Comment (elad335) [reply rpcs3/util/serialization_ext.cpp:1256]: nvm lol

**Pull request merged**